src/odemis/driver/orsay.py

```diff
@@ -2648,30 +2681,324 @@ class Scanner(model.Emitter):
+        return preset_data
+
+    def getPresetSetting(self, preset, sub_comp, setting, tag="Target"):
+        """
+        Get a setting from a preset XML ElementTree
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :param sub_comp (str): Name of the sub component
+        :param setting (str): Name of the parameter with the setting
+        :param tag (str): tag specifying the value to obtain (Most likely 'Target' or for example 'Max', 'Min',
+        'AtTarget', 'Tolerance')
+        :return (None/Str): Value found for the setting and None if no value is found.
+        """
+        for c in preset:
+            # Only interested in presets which actually defined the right component
+            if c.get("Name") == sub_comp:
+                for s in c:
+                    if s.get("Name") == setting:
+                        for t in s:
+                            if t.tag == tag:
+                                return t.text
+        else:
+            logging.warning(f"Did not find any value for the preset {preset}, with component {sub_comp} "
+                            f"and the setting {setting} using the tag {tag}.")
+            return None
+
+    def _getApertureNmbrFromPreset(self, preset):
+        """
+        Get the aperture number from a given preset.
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :return (str): Value found for the setting
+        :raises LookupError if no condenser voltage is found for a preset
+        """
+        aperture_number = self.getPresetSetting(preset, 'HybridAperture', "SelectedDiaph", tag="Target")
+        if aperture_number:
+            return int(aperture_number)
+        else:
+            raise LookupError(f"No aperture number preset found, None type was returned.")
+
+    def _getCondenserVoltageFromPreset(self, preset):
+        """
+        Get the condenser voltage from a given preset.
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :return (str): Value found for the setting
+        :raises LookupError if no condenser voltage is found for a preset
+        """
+        condenser_voltage = self.getPresetSetting(preset, 'HVPSFloatingIon', "CondensorVoltage", tag="Target")
+        if condenser_voltage:
+            return float(condenser_voltage)
+        else:
+            raise LookupError(f"No condenser voltage preset found, None type was returned.")
+
+    def _listenerImageFormat(self, image_format):
+        scale = (self.shape[0] / image_format[0],
+                self.shape[1] / image_format[1])
+        self.scale._value = scale  # Don't call the setter
+        self.scale.notify(scale)  # Do inform all the subscribers.
```

Comment:
Let's only do it if the value is different: put this inside a if-block  `if scale != self.scale.value`.